packages/editor/src/components/block-list/index.js

```diff
@@ -78,9 +80,14 @@ class BlockList extends Component {
 			this.props.onStartMultiSelect();
 		}
 
-		const boundaries = this.nodes[ this.selectionAtStart ].getBoundingClientRect();
-		const y = clientY - boundaries.top;
-		const key = findLast( this.coordMapKeys, ( coordY ) => coordY < y );
+		const blockContentBoundaries = getBlockDOMNode( this.selectionAtStart ).getBoundingClientRect();
+		if ( clientY >= blockContentBoundaries.top && clientY <= blockContentBoundaries.bottom ) {
```

Comment:
I get what this means after reading the issue but maybe it would help to leave a comment explaining why we check. 🤷‍♂️